src/filter_iterator.rs

```diff
@@ -1,15 +1,20 @@
+use async_trait::async_trait;
+
 use crate::iter::KeyValueIterator;
 use crate::types::RowEntry;
 use crate::utils::is_not_expired;
 use crate::SlateDBError;
 
 pub(crate) struct FilterIterator<T: KeyValueIterator> {
     iterator: T,
-    predicate: Box<dyn Fn(&RowEntry) -> bool + Send>,
+    predicate: Box<dyn Fn(&RowEntry) -> bool + Send + Sync>,
```

Comment:
nit: would probably be worth pulling this out into a type like:
`type FilterPredicate = Box<dyn Fn(&RowEntry) -> bool + Send + Sync>;`

Reply:
fixed in 5849e0d